### ISSUE TITLE
Type is not detected in structure

### ISSUE THREAD

**Issue by mili-l** [intellij-rust/intellij-rust#8210]
## Environment

* **IntelliJ Rust plugin version:** 0.4.162.4280-213-nightly
* **Rust toolchain version:** 1.59.0-nightly (db9d361a4 2021-11-28) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2021.3 RC Ultimate Edition (IU-213.6206)
* **Operating system:** macOS 11.5.1
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
Type is not detected in structure

## Steps to reproduce
There is a piece of code
```rust
struct N;

struct Foo<const N: usize = 1, T = N>(T);

impl Foo {
    fn new() -> Self {
        Foo(N)
    }
}

fn main() {
    let Foo::<1, N>(N) = Foo::new();
}
```
or

```rust
struct S(X);
struct X;
fn main() {
    let S(X) = S(X);
}
```
## Actual Results
`X` in ` let S(X)` is not defined and highlighted as warning
<img width="352" alt="Screenshot 2021-12-01 at 16 31 52" src="https://user-images.githubusercontent.com/1213122/144263668-f65d589b-bd90-4cbc-8595-d97026e3a84f.png">
